sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java

```diff
@@ -462,4 +464,28 @@ public CreateQueueOptions setUserMetadata(String userMetadata) {
         this.userMetadata = userMetadata;
         return this;
     }
+
+    /**
+     * Get the maxMessageSizeInKilobytes property: The maximum size of a message in kilobytes.
+     *
+     * @return the maxMessageSizeInKilobytes value.
+     */
+    public long getMaxMessageSizeInKilobytes() {
+        return this.maxMessageSizeInKilobytes;
+    }
+
+    /**
+     * Set the maxMessageSizeInKilobytes property: Represents the default maximum message size (in kilobytes)
+     * Option only available in premium tier. Default maximum in Standard tier is 256 KB, and 1 MB in premium tier.
+     * Larger message sizes are available in preview.
+     * Please see <a href=https://docs.microsoft.com/azure/service-bus-messaging/service-bus-premium-messaging#large-messages-support-preview>more info</a>
+     *
+     * @param maxMessageSizeInKilobytes the maxMessageSizeInKilobytes value to set.
+     *
+     * @return the QueueDescription object itself.
+     */
+    public CreateQueueOptions setMaxMessageSizeInKilobytes(long maxMessageSizeInKilobytes) {
```

Comment:
Do we worry about the different 'in*' methods in this class (inKilobytes and inMegabytes)? (It seems unlikely to be a concern, but I just wanted to double-check)

Reply:
Thanks Jonanthan! Maybe in the future, currently message sizes are capped to 100MB on the premium tier, so we should okay here keeping to the convention of using kB